src/common/allocator.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ============================================================================
+#pragma once
+#include <cstdio>
+#include <cstdlib>
+#include <sys/mman.h>
+
+namespace xft {
+
+constexpr size_t g_thp_threshold = (size_t)2 * 1024 * 1024;
+
+static inline bool thp_enabled() {
+    const char *str = std::getenv("ENABLE_THP");
+    return str != nullptr ? std::atoi(str) : 0;
```

Comment:
Add this ENV in environment.h?

Reply:
Better. We need to search all ENV and move it into environment.h.